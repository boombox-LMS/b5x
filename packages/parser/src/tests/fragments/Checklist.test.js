const {
  exampleChecklistMarkup,
  ChecklistDataSchema,
} = require("../../../dist/componentBuilders/fragments/parsingClasses/Checklist");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Checklist fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleChecklistMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed checklist fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed checklist fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });

  test("The parsed checklist fragment uses the expected data schema", () => {
    const validator = () => {
      ChecklistDataSchema.parse(rawFragment.data);
    };
    expect(validator).not.toThrowError();
  });
});
