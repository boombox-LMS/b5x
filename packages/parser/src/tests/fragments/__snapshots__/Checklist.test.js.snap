// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Checklist fragment builds as expected The parsed checklist fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
    "test-fragment-uri",
    "test-fragment-uri",
  ],
  "contentType": "Checklist",
  "contents": "",
  "data": Object {
    "completionContent": "You're finished! Nice work.",
  },
  "dependencyUris": Array [],
  "digest": "4d68062cfc6ff55be519e7924be24858",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": true,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
